earth2studio/models/px/dlesym.py

```diff
@@ -0,0 +1,1077 @@
+        """Load prognostic from package"""
+
+        cfg_file = Path(package.resolve("config.yaml"))
+        cfg = OmegaConf.load(cfg_file)
+        nside = cfg.data.nside
+
+        atmos_model_ckpt = package.resolve(
+            cfg.models.atmos_model_checkpoints[atmos_model_idx]
+        )
+        ocean_model_ckpt = package.resolve(
+            cfg.models.ocean_model_checkpoints[ocean_model_idx]
+        )
+
+        atmos_model = Module.from_checkpoint(atmos_model_ckpt)
+        ocean_model = Module.from_checkpoint(ocean_model_ckpt)
+        atmos_model.output_time_dim = len(cfg.io.atmos_output_times)
+        ocean_model.output_time_dim = len(cfg.io.ocean_output_times)
+
+        # Normalization constants
+        ctr = np.array(
+            [cfg.data.scaling[var]["mean"] for var in cfg.io.atmos_variables]
+            + [cfg.data.scaling[var]["mean"] for var in cfg.io.ocean_variables]
+        )
+        scl = np.array(
+            [cfg.data.scaling[var]["std"] for var in cfg.io.atmos_variables]
+            + [cfg.data.scaling[var]["std"] for var in cfg.io.ocean_variables]
+        )
+        center = ctr[None, None, None, :, None, None, None]
+        scale = scl[None, None, None, :, None, None, None]
+
+        # Constant fields
+        hpx_lat = np.load(package.resolve("hpx_lat.npy"))
+        hpx_lon = np.load(package.resolve("hpx_lon.npy"))
+        atmos_constants = np.stack(
+            [
+                np.load(package.resolve(f"{const}.npy"))
+                for const in cfg.io.atmos_constants
+            ],
+            axis=1,
+        )
+        ocean_constants = np.stack(
+            [
+                np.load(package.resolve(f"{const}.npy"))
+                for const in cfg.io.ocean_constants
+            ],
+            axis=1,
+        )
+
+        return cls(
+            atmos_model,
+            ocean_model,
+            center=center,
+            scale=scale,
+            atmos_constants=atmos_constants,
+            ocean_constants=ocean_constants,
+            hpx_lat=hpx_lat,
+            hpx_lon=hpx_lon,
+            nside=nside,
+            atmos_input_times=np.array(
+                cfg.io.atmos_input_times, dtype="timedelta64[h]"
+            ),
+            ocean_input_times=np.array(
+                cfg.io.ocean_input_times, dtype="timedelta64[h]"
+            ),
+            atmos_output_times=np.array(
+                cfg.io.atmos_output_times, dtype="timedelta64[h]"
+            ),
+            ocean_output_times=np.array(
+                cfg.io.ocean_output_times, dtype="timedelta64[h]"
+            ),
+            atmos_variables=cfg.io.atmos_variables,
+            ocean_variables=cfg.io.ocean_variables,
+            atmos_coupling_variables=cfg.io.atmos_coupling_variables,
+            ocean_coupling_variables=cfg.io.ocean_coupling_variables,
+        )
+
+    def prepare_input_data(
+        self, x: torch.Tensor, coords: CoordSystem
+    ) -> tuple[torch.Tensor, torch.Tensor]:
```

Comment:
doesnt this return a tuple[list[Tensor], list[Tensor]] or did I miss the concat?